stats/json/collector.go

```diff
@@ -83,9 +98,20 @@ func (c *Collector) Init() error {
 func (c *Collector) SetRunStatus(status lib.RunStatus) {}
 
 func (c *Collector) Run(ctx context.Context) {
-	logrus.WithField("filename", c.fname).Debug("JSON: Writing JSON metrics")
-	<-ctx.Done()
-	_ = c.outfile.Close()
+	logrus.Debug("InfluxDB: Running!")
+	ticker := time.NewTicker(time.Millisecond * 100)
+	defer func() {
+		_ = c.closeFn()
+	}()
+	for {
+		select {
+		case <-ticker.C:
```

Comment:
Given the fact that, with good reasons, we have this basic pattern in every output/collector now, we probably should think of a way to avoid the code duplication when we refactor the interface... (https://github.com/loadimpact/k6/issues/1075)

Reply:
I was thinking the same thing while writing it a second time in two days ...